src/xiaomi_roborock_vacuum.ts

```diff
@@ -1,482 +1,122 @@
-"use strict";
+import type { Config } from "./types";
+import type { Logging } from "homebridge";
+import * as events from "events";
+import { safeCall } from "./lib/safeCall";
+import { callbackify as callbackifyLib } from "./lib/callbackify";
+import type { ModelDefinition } from "./models";
+import { MODELS } from "./models";
+import semver from "semver";
 
-const semver = require("semver");
 const miio = require("./miio");
-const util = require("util");
-const callbackifyLib = require("./lib/callbackify");
-const safeCall = require("./lib/safeCall");
 
-const noop = () => {};
-
-let homebrideAPI, Service, Characteristic;
-
-const PLUGIN_NAME = "homebridge-xiaomi-roborock-vacuum";
-const ACCESSORY_NAME = "XiaomiRoborockVacuum";
-
-const MODELS = require("./models");
+const noop = () => void 0;
 const GET_STATE_INTERVAL_MS = 30000; // 30s
 
-module.exports = function (homebridge) {
-  // Accessory = homebridge.platformAccessory;
-  Service = homebridge.hap.Service;
-  Characteristic = homebridge.hap.Characteristic;
-  homebrideAPI = homebridge;
-  // UUIDGen = homebridge.hap.uuid;
-
-  homebridge.registerAccessory(
-    PLUGIN_NAME,
-    ACCESSORY_NAME,
-    XiaomiRoborockVacuum
-  );
-};
-
-class XiaomiRoborockVacuum {
+export class XiaomiRoborockVacuum extends events.EventEmitter {
```

Comment:
Maybe it's worth exploring the option of having a class for each "Service" instead of a massive class with all the logic.